File: OP/lab4/lab4 Python/QuadraticFunction.py
from TFunction import TFunction


class QuadraticFunction(TFunction):

    def calculate(self, x):
        return self._a * x ** 2 + self._b * x + self._c

    def print(self):
        if self._a == 0:
            print("y = ", end="")
        elif self._a == 1:
            print("y = x^2", end="")
        elif self._a == -1:
            print("y = -x", end="")
        else:
            print(f"y = {self._a}x^2", end="")

        if self._b == 1:
            print(" + x", end="")
        elif self._b == -1:
            print(" - x", end="")
        elif self._b != 0:
            if self._a != 0:
                if self._b > 0:
                    print(f" + {self._b}x", end="")
                else:
                    print(f" - {abs(self._b)}x", end="")
            else:
                print(f" {self._b}x", end="")

        if self._a == 0 and self._b == 0:
            print(self._c)
        elif self._c > 0:
            print(f" + {self._c}")
        elif self._c < 0:
            print(f" - {abs(self._c)}")

File: OP/lab4/lab4 C++/lab4 C++/QuadraticFunction.cpp
#include "QuadraticFunction.h"
void QuadraticFunction::print()
{
    if (a == 0)
        cout << "y = ";
    else if (a == 1)
        cout << "y = x^2";
    else if (a == -1)
        cout << "y = -x^2";
    else
        cout << "y = " << a << "x^2";


    if (b == 1)
        cout << " + x";
    else if (b == -1)
        cout << " - x";
    else if (b != 0 )
    {
        if (a != 0)
        {
            if (b > 0)
                cout << " + " << b << "x";
            else
                cout << " - " << abs(b) << "x";
        }
        else
        {
            cout << " " << b << "x";
        }        
    }

    if (a == 0 && b == 0)
        cout << c;
    else if (c > 0)
        cout << " + " << c;
    else if (c < 0)
        cout << " - " << abs(c);
    

    cout << endl;
}
double QuadraticFunction::calculate(double x)
{
    return a * pow(x, 2) + b * x + c;
}